adala/runtimes/_litellm.py

```diff
@@ -191,43 +92,26 @@ def record_to_record(
         """
 
         extra_fields = extra_fields or {}
-        field_schema = field_schema or {}
 
-        options = {}
-        for field, schema in field_schema.items():
-            if schema.get('type') == 'array':
-                options[field] = schema.get('items', {}).get('enum', [])
+        response_model = parse_template_to_pydantic_class(
+            output_template,
+            provided_field_schema=field_schema
+        )
 
-        output_fields = parse_template(
-            partial_str_format(output_template, **extra_fields),
-            include_texts=True,
+        response = get_llm_response(
+            user_prompt=input_template.format(**record, **extra_fields),
+            system_prompt=instructions_template,
+            instruction_first=instructions_first,
+            model=self.model,
+            api_key=self.api_key,
+            max_tokens=self.max_tokens,
+            temperature=self.temperature,
+            response_model=response_model
         )
-        system_prompt = instructions_template
-        user_prompt = input_template.format(**record, **extra_fields)
-        messages = [{'role': 'system', 'content': system_prompt}]
-
-        outputs = {}
-        for output_field in output_fields:
-            if output_field['type'] == 'text':
-                if user_prompt is not None:
-                    user_prompt += f"\n{output_field['text']}"
-                else:
-                    user_prompt = output_field['text']
-            elif output_field['type'] == 'var':
-                name = output_field['text']
-                messages.append({'role': 'user', 'content': user_prompt})
-                completion_text = self.execute(messages)
-                if name in options:
-                    completion_text = match_options(
-                        completion_text, options[name]
-                    )
-                outputs[name] = completion_text
-                messages.append(
-                    {'role': 'assistant', 'content': completion_text}
-                )
-                user_prompt = None
 
-        return outputs
+        response.update(**response.get('data', {}))
+        response.pop('data', None)
```

Comment:
I see this is flattening `data`, why?

Reply:
To comply with the current schema where each field in the output template makes out individual dataframe column, like `Output 1: {output1}, output 2: {output2}` -> output1, output2 columns